lib/src/frontegg_method_channel.dart

```diff
@@ -3,55 +3,95 @@ import "package:flutter/services.dart";
+  // Method channel and event channel names
   static const String methodChannelName = "frontegg_flutter";
   static const String stateEventChannelName = "frontegg_flutter/state_stream";
 
+  // Method names for platform-specific method calls
   static const String loginMethodName = "login";
   static const String logoutMethodName = "logout";
   static const String switchTenantMethodName = "switchTenant";
   static const String directLoginActionMethodName = "directLoginAction";
   static const String refreshTokenMethodName = "refreshToken";
   static const String getConstantsMethodName = "getConstants";
 
+  /// MethodChannel used for invoking platform-specific methods.
   @visibleForTesting
   final methodChannel = const MethodChannel(methodChannelName);
 
+  /// EventChannel used for listening to state changes from the native platform.
   @visibleForTesting
   final stateChangedEventChanel = const EventChannel(stateEventChannelName);
 
+  /// A stream that provides broadcast events from the native platform
+  /// related to state changes.
   late final Stream<dynamic> _stateChangedEventChanelStream =
-      stateChangedEventChanel.receiveBroadcastStream();
+  stateChangedEventChanel.receiveBroadcastStream();
 
+  /// Stream to listen to state changes from the native platform.
   @override
   Stream get stateChanged => _stateChangedEventChanelStream;
 
+  /// Initiates the login process by invoking the native platform's login method.
+  ///
+  /// Returns a [Future] that completes when the login process is finished.
   @override
   Future<void> login() => methodChannel.invokeMethod(loginMethodName);
 
+  /// Switches the current tenant by invoking the native platform's switchTenant method.
+  ///
+  /// [tenantId]: The ID of the tenant to switch to.
+  ///
+  /// Returns a [Future] that completes when the tenant switch process is finished.
   @override
   Future<void> switchTenant(String tenantId) => methodChannel.invokeMethod(
-        switchTenantMethodName,
-        {
-          "tenantId": tenantId,
-        },
-      );
+    switchTenantMethodName,
+    {
+      "tenantId": tenantId,
+    },
+  );
 
+  /// Performs a direct login action by invoking the native platform's directLoginAction method.
+  ///
+  /// [type]: The type of login action to perform.
+  /// [data]: The data associated with the login action.
+  /// [ephemeralSession]: Optional boolean indicating whether the session should be ephemeral (not sharing session with default browser). Defaults to true.
+  ///
+  /// Returns a [Future] that completes when the login action is finished.
   @override
-  Future<void> directLoginAction(String type, String data) => methodChannel.invokeMethod(
+  Future<void> directLoginAction(String type, String data,
+      {bool ephemeralSession = true}) =>
+      methodChannel.invokeMethod(
         directLoginActionMethodName,
         {
           "type": type,
           "data": data,
+          "ephemeralSession": ephemeralSession,
```

Comment:
Do we need to use the `ephemeralSession` param on iOS side(ios/Classes/FronteggMethodCallHandler.swift.directLoginAction)?

Reply:
Also, add the `ephemeralSession` to lib/src/frontegg_flutter.dart.directLoginAction so the customer developer can use this param. And make this params by default([bool ephemeralSession = true] or {bool ephemeralSession = true} )

Reply:
It's already have default value

Reply:
Yep.
We have a default value for `FronteggMethodChannel`. But users gonna use FronteggFlutter().directLoginAction, not FronteggMethodChannel().directLoginAction. 
If we don't wanna users to set up `ephemeralSession` for `directLoginAction` it is ok. But if we wanna users set up `ephemeralSession` we must add `ephemeralSession` to lib/src/frontegg_flutter.dart and set it on FronteggMethodCallHandler.swift.directLoginAction

Reply:
fixed